transport/owr_transport_agent.c

```diff
@@ -3692,3 +3852,243 @@ gchar * owr_transport_agent_get_dot_data(OwrTransportAgent *transport_agent)
+        else if ((seq > scream_rx->highest_seq) && (seq - scream_rx->highest_seq > 20000))
+            tmp_highest_seq += 65536;
+
+        /* in order */
+        if (tmp_seq >= tmp_highest_seq) {
+            diff = tmp_seq - tmp_highest_seq;
+            if (diff) {
+                if (diff >= 16)
+                    scream_rx->ack_vec = 0x0000; /* ack_vec can be reduced to guint16, initialize with 0xffff */
+                else {
+                    // Fill with potential zeros
+                    scream_rx->ack_vec = scream_rx->ack_vec >> diff;
+                    // Add previous highest seq nr to ack vector
+                    scream_rx->ack_vec = scream_rx->ack_vec | (1 << (16 - diff));
+                }
+            }
+
+            scream_rx->highest_seq = seq;
+        } else { /* out of order */
+            diff = tmp_highest_seq - tmp_seq;
+            if (diff < 16)
+                scream_rx->ack_vec = scream_rx->ack_vec | (1 << (16 - diff));
+        }
+        if (!(scream_rx->ack_vec & (1 << (16-5)))) {
+            /*
+            * Detect lost packets with a little grace time to cater
+            * for out-of-order delivery
+            */
+            scream_rx->n_loss++; /* n_loss is a guint8 , initialize to 0 */
+        }
+
+        /*
+        * ECN is not implemented but we add this just to not forget it
+        * in case ECN flies some day
+        */
+        scream_rx->n_ecn = 0;
+        scream_rx->last_feedback_wallclock = (guint32)(arrival_time / 1000000);
+        rtcp_info = g_hash_table_new(g_str_hash, g_str_equal);
+        g_hash_table_insert(rtcp_info, "pt", GUINT_TO_POINTER(GST_RTCP_TYPE_RTPFB));
+        g_hash_table_insert(rtcp_info, "fmt", GUINT_TO_POINTER(GST_RTCP_RTPFB_TYPE_SCREAM));
+        g_hash_table_insert(rtcp_info, "ssrc", GUINT_TO_POINTER(ssrc));
+        g_hash_table_insert(rtcp_info, "last-feedback-wallclock",
+            GUINT_TO_POINTER(scream_rx->last_feedback_wallclock));
+        g_hash_table_insert(rtcp_info, "highest-seq",
+            GUINT_TO_POINTER(scream_rx->highest_seq));
+        g_hash_table_insert(rtcp_info, "n-loss", GUINT_TO_POINTER(scream_rx->n_loss));
+        g_hash_table_insert(rtcp_info, "n-ecn", GUINT_TO_POINTER(scream_rx->n_ecn));
+        g_hash_table_insert(rtcp_info, "session-id", GUINT_TO_POINTER(session_id));
+
+        g_mutex_lock(&priv->rtcp_lock);
+        it = g_list_find_custom(priv->rtcp_list, (gpointer)rtcp_info,
+            (GCompareFunc)compare_rtcp_scream);
+
+        if (it) {
+            g_hash_table_unref((GHashTable *)it->data);
+            priv->rtcp_list = g_list_delete_link(priv->rtcp_list, it);
+        }
+        priv->rtcp_list = g_list_append(priv->rtcp_list, rtcp_info);
+        OWR_UNUSED(it);
+        g_mutex_unlock(&priv->rtcp_lock);
+        g_signal_emit_by_name(priv->rtpbin, "get-internal-session", session_id, &rtp_session);
+        g_signal_emit_by_name(rtp_session, "send-rtcp", 20000000);
```

Comment:
Yes, actually "send-rtcp-full". No idea if and how you could use it, maybe the scream algorithm has some use of that information :) I just mentioned it in case it is useful to have